cmd/test/main.go

```diff
@@ -0,0 +1,368 @@
+		}
+		fmt.Printf("Deleting database %s\n", name)
+		cmd2 := ravendb.NewDeleteDatabaseCommand(name, true, "")
+		res2, err := ravendb.ExecuteDeleteDatabaseCommand(exec, cmd2)
+		must(err)
+		panicIf(res2.RaftCommandIndex == 0, "res2.RaftCommandIndex is 0")
+		if verboseLog {
+			fmt.Printf("res2: %#v\n", res2)
+		}
+	}
+}
+
+func testDeleteDatabaseOp() {
+	exec := getExecutor()
+	cmd := ravendb.NewDeleteDatabaseCommand(testDbName, true, "")
+	res, err := ravendb.ExecuteDeleteDatabaseCommand(exec, cmd)
+	must(err)
+	panicIf(res.RaftCommandIndex == 0, "res.RaftCommandIndex is 0")
+	if verboseLog {
+		fmt.Printf("res: %#v\n", res)
+	}
+
+	fmt.Printf("testDeleteDatabaseCommand ok\n")
+}
+
+/*
+func testCreateAndDeleteDatabaseCommand() {
+	dbName := "tst_" + ravendb.NewUUID().Hex()
+	exec := getExecutor()
+	cmd := ravendb.NewCreateDatabaseCommand(dbName, 1)
+	res, err := ravendb.ExecuteCreateDatabaseCommand(exec, cmd)
+	must(err)
+	panicIf(res.RaftCommandIndex == 0, "res.RaftCommandIndex is 0")
+	panicIf(res.Name != dbName, "res.Name is '%s', expected '%s'", res.Name, dbName)
+	if true || verboseLog {
+		fmt.Printf("res: %#v\n", res)
+	}
+
+	// TODO: do I need to wait?
+
+	cmd2 := ravendb.NewDeleteDatabaseCommand(dbName, false, "")
+	res2, err := ravendb.ExecuteDeleteDatabaseCommand(exec, cmd2)
+	must(err)
+	panicIf(res2.RaftCommandIndex == 0, "res2.RaftCommandIndex is 0")
+	if verboseLog {
+		fmt.Printf("res2: %#v\n", res2)
+	}
+
+	fmt.Printf("testCreateAndDeleteDatabaseCommand ok\n")
+}
+*/
+
+func testPutGetDeleteDocument() {
+	sess := mustGetSession()
+	exec := sess.RequestExecutor.GetCommandExecutor(false)
+	key := "testing/1"
+	meta := map[string]interface{}{
+		"@collection": "Testings",
+	}
+	doc := map[string]interface{}{
+		"Name":      "test1",
+		"DocNumber": 1,
+		"@metadata": meta,
+	}
+	cmd := ravendb.NewPutDocumentRawCommand(key, doc, "")
+	res, err := ravendb.ExecutePutDocumentRawCommand(exec, cmd)
+	must(err)
+	if verboseLog {
+		fmt.Printf("res: %#v\n", res)
+	}
+
+	cmd = ravendb.NewGetDocumentCommand([]string{key}, nil, false)
+	res2, err := ravendb.ExecuteGetDocumentCommand(exec, cmd)
+	must(err)
+	if verboseLog {
+		fmt.Printf("len(res2.Includes): %d, len(res2.Results): %d\n", len(res2.Includes), len(res2.Results))
+	}
+
+	// test get of non-existent document
+	cmd = ravendb.NewGetDocumentCommand([]string{"testing/asdf"}, nil, false)
+	res3, err := ravendb.ExecuteGetDocumentCommand(exec, cmd)
+	panicIf(err == nil, "unexpected err is nil")
+	panicIf(res3 != nil, "unexpected res3 is != nil: %#v", res3)
+	// verify that it returns 404 Not Found
+	notFound := err.(*ravendb.NotFoundError)
+	if verboseLog {
+		fmt.Printf("not found url: '%s'\n", notFound.URL)
+	}
+
+	cmd = ravendb.NewDeleteDocumentCommand(key, "")
+	err = ravendb.ExecuteDeleteDocumentCommand(exec, cmd)
+	must(err)
+
+	// test delete of non-existent document
+	// it succeeds even if document doesn't exist
+	cmd = ravendb.NewDeleteDocumentCommand("testing/asdf", "")
```

Comment:
I assume the second argument here is the change vector? If so, the way it works is that empty change vector is "verify the doc doesn't exists", null change vector means don't do any verification and anything else does an exact match

Reply:
Didn't realize this subtlety. Unlike Java, in Go a string cannot be nil, so I'll have to rework the type used for change vector, probably to *string.